node/src/components/storage.rs

```diff
@@ -695,6 +709,50 @@ impl Storage {
         })
     }
 
+    /// Retrieves single block header by height by looking it up in the index and returning it.
+    fn get_block_header_and_finality_signatures_by_height<Tx: Transaction>(
+        &self,
+        tx: &mut Tx,
+        height: u64,
+    ) -> Result<Option<BlockHeaderAndFinalitySignatures>, Error> {
+        let block_hash = match self.block_height_index.get(&height) {
+            None => return Ok(None),
+            Some(block_hash) => block_hash,
+        };
+        let block_header = match self.get_single_block_header(tx, block_hash)? {
+            None => return Ok(None),
+            Some(block_header) => block_header,
+        };
+        let finality_signatures = self
+            .get_finality_signatures(tx, block_hash)?
+            .map_or_else(BTreeMap::new, |block_signatures| block_signatures.proofs);
+        for (public_key, signature) in finality_signatures.iter() {
+            FinalitySignature::verify_bare_finality_signature(
+                &block_hash,
+                &block_header.era_id(),
+                signature,
+                public_key,
+            )
+            .map_err(Error::CryptographicSignatureVerificationError)?
+        }
```

Comment:
I agree with @Fraser999 here – we already validate every finality signature on startup and we verify them before storing.

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/1033/commits/6b316c6752128c2fd6df8bf7bf585d680874f477